### PR TITLE
Bugfix/ls24003505/move and movel with boolean type

### PR DESCRIPTION
**Pull request by davidepalladino-apuliasoft**
## Description
This work adds the capability to use `MOVE` and `MOVEL` when the result is an indicator.
It is assumed that for blank or decimal value as Factor 2, the final value considered will be `0`.
There are these atomic tests: `MU101016`, `MU101017`, `MU101018` and `MU101019`.
Other additional test are provided.

Related to #LS24003505

## Checklist:
- [x] There are tests regarding this feature
- [x] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [x] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/interpreter/JarikoCallbackTest.kt [hunk 1]
```diff
@@ -566,6 +566,48 @@ class JarikoCallbackTest : AbstractTest() {
         executeSourceLineTest("ERROR27")
     }
 
+    @Test
+    fun executeERROR29CallBackTest() {
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/interpreter/JarikoCallbackTest.kt:570]: For each of this test you should create `executeERRORnnSourceLineTest()`

**Review by lanarimarco — CHANGES_REQUESTED** [review]

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt [hunk 2]
```diff
@@ -114,6 +117,24 @@ fun move(
...
+private fun getStringValueToMove(
+    target: AssignableExpression,
+    interpreterCore: InterpreterCore,
+    value: Expression
+) = when {
+    target.type() is BooleanType -> {
+        val valueInterpreted: String = (interpreterCore.eval(value)).asString().value
+        when {
+            (valueInterpreted.isInt() && valueInterpreted.toInt() in 0..1) -> if (valueInterpreted.toInt() == 0) "0" else "1"
+            (valueInterpreted.isDecimal() && valueInterpreted.toDecimal() in 0.0..1.0) -> "0"
```
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt:129]: For RPG is that right. I was shocked while I was testing on AS400.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt [hunk 3]
```diff
@@ -114,6 +117,24 @@ fun move(
...
 
+private fun getStringValueToMove(
+    target: AssignableExpression,
+    interpreterCore: InterpreterCore,
+    value: Expression
+) = when {
+    target.type() is BooleanType -> {
+        val valueInterpreted: String = (interpreterCore.eval(value)).asString().value
+        when {
+            (valueInterpreted.isInt() && valueInterpreted.toInt() in 0..1) -> if (valueInterpreted.toInt() == 0) "0" else "1"
```
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt]: I agree, it is a typo

**Review by lanarimarco — APPROVED** [review]
ok

**Pull request merged**